### DISCUSSION TITLE
Put Pollen & Mold details on multiple lines to avoid ellipsis

### DISCUSSION THREAD

**Discussion by nimxaa** [breezy-weather/breezy-weather#972] [Ideas]
### Steps to reproduce

1. View "Pollen & mold" section.

### Expected behavior

To see full text.

### Actual behavior

Text being cut off.

### Weather source used

AccuWeather

### Breezy Weather version

4.6.10-beta

### Android version

Android 11

### Device

OnePlus 7T Pro

### Other details

<img src="https://github.com/breezy-weather/breezy-weather/assets/145933422/9b294ebc-2eaf-4575-aa90-ba6d7dad9950" width="50%">

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I read the [Frequently Asked Questions](https://github.com/breezy-weather/breezy-weather/blob/main/HELP.md) document and followed instructions if relevant.
- [X] I have updated the app to **[version 4.6.9 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance), unless my issue only happens with a specific language
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Cut off depends on the language and is intended to not overlap.
"Very low" is redundant with the color dot, so it's not like you're missing anything important.
What's your suggestion of improvement?

**Comment by nimxaa** [comment]
Would it look too bad if it just were in new line?

**Comment by papjul** [comment]
Needs some tests (probably best during pollen season)

**Comment by min7-i** [comment]
In my opinion, increasing the maximum number of lines to two for the subtitle would be a viable solution. To also account for very large font/display scales we could additionally limit the number of items per row to one when a certain threshold is crossed.

| 1 | 2 |
| --- | --- |
| ![line break](https://github.com/user-attachments/assets/3a965578-0be1-43a2-998b-695a66e0385c) | ![items per row](https://github.com/user-attachments/assets/c962c17b-6558-4d73-a06c-409afcf2a5d6) |

In #996 it was also suggested to use a marquee text, but I'd prefer to have a solution where the subtitle is fully visible without any animation. At most, I could image using that for the title (pollen name) and only start the marquee animation when tapping on the text.

**Reply by min7-i** [reply]
> In my opinion, increasing the maximum number of lines to two for the subtitle would be a viable solution. To also account for very large font/display scales we could additionally limit the number of items per row to one when a certain threshold is crossed.

This would still be a viable approach for the new pollen details screen and the home fragment in my opinion. Unless you're planning to replace this entirely with a chart, @papjul.

**Reply by papjul** [reply]
I would prefer a solution 2 which auto-detects when either the pollen name or the concentration/risk level has a width larger than 50 % of the available space, and puts everything on 100 % of the available space in that case.

I don't know if it's feasible?

**Reply by min7-i** [reply]
Initially, I used the density (display scale) and text scale as thresholds, but that only works if the defaults aren't used and won't cover all cases. Measuring the width of the texts would certainly be more reliable. I'll check if there's an efficient way to do this.
